src/GraphQL/GraphQLBuilderExtensions.cs

```diff
@@ -721,37 +721,48 @@ public static IGraphQLBuilder AddDocumentCache<TDocumentCache>(this IGraphQLBuil
         }
         #endregion
 
-        #region - AddDocumentWriter -
+        #region - AddSerializer -
         /// <summary>
-        /// Registers <typeparamref name="TDocumentWriter"/> as a singleton of type <see cref="IDocumentWriter"/> within the
+        /// Registers <typeparamref name="TSerializer"/> as a singleton of type <see cref="IGraphQLSerializer"/> within the
         /// dependency injection framework.
+        /// If supported, the class is also registered as type <see cref="IGraphQLTextSerializer"/>.
         /// </summary>
-        public static IGraphQLBuilder AddDocumentWriter<TDocumentWriter>(this IGraphQLBuilder builder)
-            where TDocumentWriter : class, IDocumentWriter
+        public static IGraphQLBuilder AddSerializer<TSerializer>(this IGraphQLBuilder builder)
+            where TSerializer : class, IGraphQLSerializer
         {
-            builder.Services.Register<IDocumentWriter, TDocumentWriter>(ServiceLifetime.Singleton, true);
+            builder.Services.Register<IGraphQLSerializer, TSerializer>(ServiceLifetime.Singleton, true);
+            if (typeof(IGraphQLTextSerializer).IsAssignableFrom(typeof(TSerializer)))
+                builder.Services.Register(typeof(IGraphQLTextSerializer), typeof(TSerializer), ServiceLifetime.Singleton, true);
+                //builder.Services.Register(services => (IGraphQLTextSerializer)services.GetRequiredService<IGraphQLSerializer>(), ServiceLifetime.Singleton);
```

Comment:
Prep for subsequent PR to have one service pull from the other as we previously discussed.  I just want to discuss and change it all at once rather than piecemeal.

Reply:
Discuss what exaclty? Pull the same object or not?

Reply:
See:
- #2797